### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by rotsen18**

### CODE REVIEW

File: app/customers.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+
+class Customer:
+    def __init__(self, user_info, fuel_price):
+        self.name = user_info["name"]
+        self.product_cart = user_info["product_cart"]
+        self.location = user_info["location"]
+        self.money = user_info["money"]
+        self.car = user_info["car"]
+        self.fuel_price = fuel_price
```
Comment (Magus6784) [inline app/customers.py:11]: I don't think fuel price is correct attribute for Customer. You know, fuel price not connect to Customer. If fuel price changed you need to change all your Customer

File: app/customers.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+    def fuel_costs(self, shop):
+        distance = math.dist(shop.location, self.location)
+        return self.fuel_price * distance * self.car["fuel_consumption"] / 100
+
+    def pay_for_trip(self, amount):
+        self.money -= amount
+        return self.money
```
Comment (Magus6784) [inline app/customers.py:19]: You don't need to return self.money. You can just use customer.money in main/$20

File: app/main.py [hunk 3]
```diff
@@ -1,3 +1,57 @@
...
+    for shop in shops:
+        costs = customer.trip_costs(shop)
+        print(f"{customer.name}'s trip to the {shop.name} costs {costs}")
+        shops_cost[costs] = shop
+    shops_cost = dict(sorted(shops_cost.items()))
+
+    for costs, shop in shops_cost.items():
+        if costs <= customer.money:
+            return shop, costs
+    return None, None
```
Comment (Magus6784) [inline app/main.py:39]: maybe its better to create method for Customer.
If you create method for trip in Customer, why choose shop not in class too?

**Review by Magus6784 — CHANGES_REQUESTED** [review]

**Comment by Magus6784** [comment]
Also i think better name for files is just. "customer" and "shop". Like in tasl elfes and dwarfes we use only "Elf" "Dwarf"

**Comment by Magus6784** [comment]
And a little more
"When the customer arrives at the shop his location should equal to shop location."

**Review by TPolina — APPROVED** [review]
well done!

**Pull request closed without merging**